### PR TITLE
Make IGV filtering instant (SCP-5662)

### PR DESCRIPTION
**Pull request by eweitz**
This drastically speeds up IGV filtering by cell annotations, making novel genome browser functionality a delight to use.

Here's how it looks!

https://github.com/broadinstitute/single_cell_portal_core/assets/1334561/c257354e-4bd9-4b67-804b-b71187daf90b

### Overview
[Previously](https://github.com/broadinstitute/single_cell_portal_core/pull/2053), filtering in IGV was new and transformative, but painfully slow.  It took 10-30 seconds to filter.  That was drastically faster than writing code, running it, and uploading new tracks. But that's still much slower than users expect in a web app.

Now, IGV filtering is instant.  This is achieved by only applying filter logic to genomic features that are in-frame, i.e. in the current genomic window in igv.js.  These genomic features are typically fully contained in the frame, but can also merely partly overlap at left (i.e. upstream) or right (downstream) of the genomic coordinates in view.

### Next steps
Filtering is fast now, but still brittle.  For example, panning to left or right or zooming out will not show genomic features that should be there.  Searching for a new gene also still makes things go awry in the genome browser.  This PR doesn't fully satisfy SCP-5662, but helps show this prototype filtering to stakeholders.  Robustness improvements are next.

### Test
An automated test was updated to account for this change.  If you'd like to manually test:
1. Go to a multiome study
2. Search a gene
3. Apply filters
4. Confirm genomic features in IGV track update, and quickly

This relates to SCP-5662.

### CODE REVIEW

File: cell-filtering-igv/index.html [hunk 1]
```diff
@@ -37,26 +34,60 @@
   return originalChrFeatures
 }
 
-  function filterAtac(scoreSelection) {
-    const trackIndex = 4 // Track index
-    const igvBrowser = window.igvBrowser
+/** Determine if feature is in genomic frame */
+function getIsFeatureInFrame(feature, igvBrowser) {
```
Comment (bistline) [inline cell-filtering-igv/index.html:38]: Why is this function defined twice, both here and in `GenomeView`?

**Review by bistline — APPROVED** [review]
Great improvement!  Very exciting to see this productionized further.

File: cell-filtering-igv/index.html (hunk 1 above)
Comment (eweitz) [reply cell-filtering-igv/index.html:38]: This area intended is as a 0-config, rapid prototyping environment.  It's super rough -- stuff is copied and pasted, poorly formatted, etc. -- but helps iterate 5-20x faster.

**Review by jlchang — APPROVED** [review]
Functional review - fragment filtering for CEBPB in 10x multiome study no longer needs page refresh to see different filtered states and filtering while visualizing SPIB seems more snappy.

**Pull request merged**